tendrl/alerting/notification/handlers/mail_handler.py

```diff
@@ -0,0 +1,283 @@
+                return True
+            else:
+                if 'email_pass' not in config:
+                    raise InvalidHandlerConfig(
+                        'email_pass is a mandatory field if auth is chosen'
+                    )
+        return True
+
+    def add_destination(self, conf):
+        config = ast.literal_eval(conf)
+        try:
+            if self.validate_config(config):
+                self.etcd_server.client.write(
+                    '/alerting/notification_medium/email/config/%s' % config[
+                        'email_id'],
+                    config
+                )
+        except (
+            NotificationDispatchError,
+            InvalidHandlerConfig,
+            EtcdConnectionFailed,
+            EtcdNotDir,
+            KeyError,
+            SyntaxError,
+            ValueError
+        ) as ex:
+            LOG.error(
+                'Failed to add email config %s. Error %s' % (conf, str(ex))
+            )
+            raise NotificationDispatchError(str(ex))
+
+    def set_destinations(self):
+        admin_config = {}
+        user_configs = []
+        try:
+            configs = self.etcd_server.client.read(
+                '/alerting/notification_medium/email/config',
+                recursive=True
+            )
+            for config in configs._children:
+                mail_config = ast.literal_eval(config['value'])
+                if mail_config['is_admin']:
+                    if mail_config.get('auth') is None:
+                        mail_config['auth'] = ""
+                    admin_config = mail_config
+                else:
+                    user_configs.append(mail_config)
+            self.admin_config = admin_config
+            self.user_configs = user_configs
+        except EtcdKeyNotFound as ex:
+            LOG.error(
+                'Could not fetch mail configuration.Error %s'
+                % str(ex)
+            )
+        except (
+            EtcdConnectionFailed,
+            EtcdNotDir,
+            ValueError,
+            KeyError,
+            SyntaxError
+        ) as ex:
+            LOG.error(
+                'Could not fetch mail configuration.Error %s'
+                % str(ex)
+            )
+            raise NotificationDispatchError(str(ex))
+
+    def format_message(self, alert):
+        return "Subject: [Alert] %s, %s threshold breached\n\n%s" % (
+            alert.resource, alert.severity, alert.to_json_string())
+
+    def set_alert_destinations(self, alert):
+        recipients = []
+        for config in self.user_configs:
+            # TODO(anmol) Ideally even check the cluster part.
+            if config['alert_subscriptions'] == '*' or\
+                    alert.resource in config['alert_subscriptions']:
+                recipients.append(config['email_id'])
+        return recipients
+
+    def __init__(self):
+        self.name = 'email'
+        try:
+            etcd_kwargs = {
```

Comment:
If this class is created from manager, the config loaded in manager should be passed as a constructor argument and assigned to an instance variable of this class and used.